clang/lib/Sema/SemaSYCL.cpp

```diff
@@ -678,30 +686,73 @@ constructKernelName(Sema &S, FunctionDecl *KernelCallerFunc,
 namespace {
 
 QualType getItemType(const FieldDecl *FD) { return FD->getType(); }
-QualType getItemType(const CXXBaseSpecifier &BS) { return BS.getType(); }
+// FIXME enable this function when base classes are supported
```

Comment:
This ends up requiring disabling other code, right?  We already had some base stuff working which is necessary and depended on this.

Reply:
Base class support isn't implemented. Leaving this function in caused a compiler warning (unused function), so I took it out until base class support is fully implemented.

Reply:
Only because you removed other functioning code.  Please don't do that.

Reply:
Please confirm that base class support for the integration header was complete and correct.

Reply:
It is at least mostly correct.  Additionally, others are working on making base classes work for everything, so removing this would inhibit their work, and I'd be against any patch that inhibited that work on those grounds.